### DISCUSSION TITLE
Context.bots数组的元素类型为Bot<never>，不存在sendGroupMsg方法

### DISCUSSION THREAD

**Discussion by werifu** [koishijs/koishi#344] [Q&A]
**Describe the bug**
`let bot = ctx.bots[0];`
`bot.sendGroupMsg('群号', '1');
`
使用yarn koishi start -- -- -r ts-node/register启动，加载插件时（.ts文件）
`TSError: ⨯ Unable to compile TypeScript:
    plugins/caker.ts:70:29 - error TS2339: Property 'sendGroupMsg' does not exist on type 'Bot<never>'.`
在vscode里点进看Context的声明也能看到`get bots(): Bot<never>[] & Record<string, Bot<never>>;`成员。
与文档中描述的Bot[]不同
**To Reproduce**

**Expected behavior**

能正常加载插件。或者有什么办法调用sendGroupMsg等bot的接口
**Screenshots**

![image](https://user-images.githubusercontent.com/53432474/126811206-254ccf8d-dbd9-464a-ad19-894c59a14c84.png)
![image](https://user-images.githubusercontent.com/53432474/126811240-078b51f4-d594-4188-aa9d-03b486b07820.png)

**Versions**
- OS: linuxx64
- Node Version: 14.17.3
- ts-node Version: 10.1.0
- Koishi Version: 3.12.3

**Additional context**
在koishi2的时候是正常的，更新到3后就这样，不知道是koishi的特性还是bug，如果是因为我打开姿势错了，希望能得到正确打开方式，打扰了。
然后想问正确的调用ctx中bot是否有推荐的方式，总感觉ctx.bot[0]是很野蛮的做法（

**Comment by NWYLZW** [comment]
ctx.getBot

**Comment by shigma** [comment]
你可以使用 `ctx.bots[0].sendMessage`

**Comment by shigma** [comment]
> 总感觉 ctx.bots[0] 是很野蛮的做法（

其实建议 `ctx.bots.map()` 或者直接 `ctx.broadcast()`，比较具有扩展性
